### DISCUSSION TITLE
2D Games?

### DISCUSSION THREAD

**Discussion by diagom02** [SlashScreen/skelerealms#25] [Q&A] [answered]
Will it not be compatible with, isometric or top-down games 2D for version 1.0?

**Comment by SlashScreen** [comment, marked as answer]
I’m not planning to add native 2D support, but I’ll try to remember to add a guide for what parts of the script to change. If memory serves, most of the 3D-dependent starts are merely some classes inheriting from node3d and the main distance calculation script in the entity manager that determines whether an entity is in the scene or not. It should be pretty simple, at the very least. I can’t speak on isometric, since I’ve never made one of those before.

**Reply by diagom02** [reply]
I'm in learning mode, so if I manage to modify those scripts that inherit from node3d could you consider adding them to the plugin?

Saludos!

**Reply by SlashScreen** [reply]
> I'm in learning mode, so if I manage to modify those scripts that inherit from node3d could you consider adding them to the plugin?
> 
> Saludos!

No, it's a fundamental change in the system that's difficult to add without preprocessor flags in GDScript, and I don't want to keep a separate copy of the script because it means I have to maintain both.
